.github/workflows/gh-pages.yml

```diff
@@ -14,6 +14,19 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: "pip"
+
+      - name: Install Python dependencies
+        run: pip install -r ./script/requirements.txt
+
+      - name: Regenerate data files from CSV
+        working-directory: ./script
+        run: python convertJSON.py
+
```

Comment:
Good point! I think it's not possible using that syntax, since that only decides whether or not the workflow should be run. We could instead do something like this: https://stackoverflow.com/a/70711156/20382305

However, I think that's a bit complicated for something that wasn't requested directly by the client. But feel free to open an `enhancement`-tagged issue with that idea!

Reply:
Ah - good point. Yes, this looks like it's going to either run the action as a whole, or not at all. I wonder what happens if we were to create two separate actions: one for the Python data conversion, one for the GH pages build. But that would probably get too messy & we might end up debugging GitHub Action race conditions... I'd say let's leave it as is, and revisit if there's ever explicit client demand.